app/controllers/admin/routes.py

```diff
@@ -157,6 +159,62 @@ def rsvpLogDisplay(eventId):
     else:
         abort(403)
 
+@admin_bp.route('/event/<eventId>/renew', methods=['POST'])
+def renewEvent(eventId):
+    try: 
+        formData=request.form
+        try:
+            assert formData['timeStart'] < formData['timeEnd']
+        except AssertionError:
+            flash("End time must be after start time", 'warning')
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+        
+        try:
+            if formData.get('dateEnd'):
+                assert formData['dateStart'] < formData['dateEnd']
+        except AssertionError:
+            flash("End date must be after start date", 'warning')
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+
+
+        priorEvent = Event.get_by_id(eventId)
+        newEvent, message = attemptSaveEvent({
```

Comment:
Get your dictionary from the prior event and update with your new form data. That new dict should be passed to `attemptSaveEvent` after its ID is removed.

Reply:
I have resolved this.